api/v1alpha1/addon_types.go

```diff
@@ -134,40 +142,158 @@ func (ClusterAutoscaler) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+type DefaultStorage struct {
+	ProviderName           string `json:"providerName"`
+	StorageClassConfigName string `json:"storageClassConfigName"`
+}
+
 type CSIProviders struct {
 	// +optional
 	Providers []CSIProvider `json:"providers,omitempty"`
 	// +optional
-	DefaultClassName string `json:"defaultClassName,omitempty"`
+	DefaultStorage *DefaultStorage `json:"defaultStorage,omitempty"`
 }
 
 type CSIProvider struct {
-	Name string `json:"name,omitempty"`
+	Name string `json:"name"`
+
+	// +optional
+	StorageClassConfig []StorageClassConfig `json:"storageClassConfig,omitempty"`
+
+	Strategy AddonStrategy `json:"strategy"`
+
+	// +optional
+	Credentials *corev1.SecretReference `json:"credentials,omitempty"`
+}
+
+type StorageClassConfig struct {
+	Name string `json:"name"`
+
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty"`
+
+	// +optional
+	ReclaimPolicy string `json:"reclaimPolicy,omitempty"`
+
+	// +optional
+	VolumeBindingMode string `json:"volumeBindingMode,omitempty"`
+}
+
+func (StorageClassConfig) VariableSchema() clusterv1.VariableSchema {
+	supportedReclaimPolicies := []string{
+		VolumeReclaimRecycle,
+		VolumeReclaimDelete,
+		VolumeReclaimRetain,
+	}
+	supportedBindingModes := []string{
+		VolumeBindingImmediate,
+		VolumeBindingWaitForFirstConsumer,
+	}
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"name": {
+					Type:        "string",
+					Description: "Name of storage class config.",
+				},
+				"parameters": {
+					Type:                   "object",
+					Description:            "Parameters passed into the storage class object.",
+					XPreserveUnknownFields: true,
+				},
+				"reclaimPolicy": {
+					Type: "string",
+					Enum: variables.MustMarshalValuesToEnumJSON(supportedReclaimPolicies...),
+				},
+				"volumeBindingMode": {
+					Type: "string",
+					Enum: variables.MustMarshalValuesToEnumJSON(supportedBindingModes...),
+				},
```

Comment:
Lets set the defaults in the API that match DKP's? These would be `VolumeBindingWaitForFirstConsumer` and `VolumeReclaimDelete`

Reply:
Added!